### PR TITLE
Feature/#1581 migrate reviews on maps and mods

### LINKED ISSUE #1348: [Feature Request] Display reviews and score for previous versions of mods
This is very much needed for mods that update periodically. As their reviews are all lost, and they can't really stand out anymore, especially compared to older mods that have not been updated in years. It kind of defeats the purpose of reviews if they just vanish when the version is iterated.

The current display hurts mods that are new or updating, since it's hard to distinguish popular, but regularly updating mods from the noise. Especially when the highest rated mods that are features on the vault are just old mods that have not seen updates for years. Uvesto is a good example of this, which I've seen passed over by both myself and friends that are new to FAF because it has no reviews, yet is fairly popular, well made, stable, regularly updated, and highly rated.

**Proposal 1:**

- Compile all review scores into the mod score in the mod vault view.
- Include all review counts as the review count in the mod vault view
- On the mod details modal, display current versions score and reviews separate from combined previous versions
  - Reviews have a small icon with the version# the review was for

**Proposal 2:**

- Display current review score and # of reviews alongside combined score and # of reviews in Mod Vault view
- On the mod details modal, display current versions score and reviews separate from combined previous versions
  - Reviews have a small icon with the version# the review was for

**Example Mod Vault:**

image

Please let me know your thoughts. I'm a bit passionate about this, since it makes finding decent mods harder than it should be. And is a bit disheartening to see your mods reviews vanish whenever it's updated...

### LINKED ISSUE #1581: migrate map rating/review
Hello,
I was wondering why the map rating/review is not migrated when u upload a new map version?
image

@axel1200:
> It is not a bug because it was intended on the other hand it is suboptimal i guess

Are there any problem I can't see? Because it would be very nice if the map rating/review could be migrated.

### PR DESCRIPTION
**Pull request by Sheikah45**
Fixes #1581 
Fixes #1348 

How the reviews look with the version label
Screenshot (162)

Showing how even recently updated versions keep the reviews.
Screenshot (161)

### CODE REVIEW

**Comment by Brutus5000** [comment]
The current version should be highlighted with an additional badge or be called "current version" or something similar

**Comment by Sheikah45** [comment]
> 
> 
> The current version should be highlighted with an additional badge or be called "current version" or something similar

Good idea

File: src/main/java/com/faforever/client/mod/ModDetailController.java [hunk 1]
```diff
@@ -147,7 +147,8 @@ public void setModVersion(ModVersion modVersion) {
     Player player = playerService.getCurrentPlayer()
         .orElseThrow(() -> new IllegalStateException("No current player is available"));
 
-    reviewsController.setCanWriteReview(modService.isModInstalled(modVersion.getUid()));
+    reviewsController.setCanWriteReview(modService.isModInstalled(modVersion.getUid())
+        && !modVersion.getMod().getAuthor().equals(player.getUsername()));
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/mod/ModDetailController.java:151]: Sneaky sneaky ;-)

**Review by Brutus5000 — APPROVED** [review]

File: src/main/java/com/faforever/client/mod/ModDetailController.java (hunk 1 above)
Comment (Sheikah45) [reply src/main/java/com/faforever/client/mod/ModDetailController.java:151]: Requested by our creative councilor

**Pull request merged**